src/Store/VimStoreConnector.re

```diff
@@ -975,6 +980,42 @@ let start =
     });
   };
 
+  let zoomStep = 0.2;
+  let defaultZoom = 1.0;
+  let minZoomValue = 0.0;
+  let maxZoomValue = 2.8;
+  let zoomEffect = (state: State.t, zoomAction) =>
+    Isolinear.Effect.create(~name="vim.zoom", () => {
+      let configuration = state.configuration;
+      let currentZoomValue =
+        Core.Configuration.getValue(c => c.uiZoom, configuration);
+
+      let calculatedZoomValue =
+        switch (zoomAction) {
+        | ZoomIn => currentZoomValue +. zoomStep
+        | ZoomOut => currentZoomValue -. zoomStep
+        | ZoomReset => defaultZoom
+        };
```

Comment:
What's the reason you pass in a `zoomAction` instead of just passing in the value directly?

Reply:
What do you mean by 
> instead of just passing in the value directly

Can't get your point.

Reply:
Looking closer, I guess you can't pass in `currentZoomValue +. zoomStep)` directly instead of `ZoomIn` because you need `currentZoomValue`. But you could do `zoomEffect(state, zoom => zoom +. zoomStep)`.